### PR TITLE
Marketplace Reviews: Show the user review first on the list

### PR DESCRIPTION
**Pull request by WBerredo**
Related to https://github.com/Automattic/wp-calypso/pull/85496

## Proposed Changes

Always display the user review first on the list.
Two requests are being made:
* User reviews for a given product (limited to 1)
* Reviews excluding the user

The lists will be concatenated, keeping this way the user review always as first.

## Testing Instructions

* On this branch
* Go to a plugin details page where you have a review. Ex: `/plugins/woocommerce-bookings/`
* On the bottom, click on Read all reviews
* Check if your review is being displayed first

CleanShot 2023-12-20 at 08 37 14@2x

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by WBerredo** [comment]
On hold until https://github.com/Automattic/wp-calypso/pull/85496 is merged.

**Review by gcsecsey — APPROVED** [review]
I like this approach, the changes look good to me and work great!

I think we can merge this once https://github.com/Automattic/wp-calypso/pull/85496 is merged.

**Pull request merged**
